pkg/operation/seed/scheduler/gardener_kube_scheduler.go

```diff
@@ -0,0 +1,399 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
+
+import (
+	"context"
+	"fmt"
+
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/seed/scheduler/configurator"
+
+	"github.com/Masterminds/semver"
+	"github.com/pkg/errors"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	name                                  = "gardener-kube-scheduler"
+	containerName                         = "kube-scheduler"
+	portNameMetrics                       = "metrics"
+	dataKeyComponentConfig                = "config.yaml"
+	kubeSchedulerClusterRoleBindingName   = "gardener.cloud:kube-scheduler"
+	volumeSchedulerClusterRoleBindingName = "gardener.cloud:volume-scheduler"
+	roleBindingName                       = "gardener.cloud:kube-scheduler:extension-apiserver-authentication-reader"
+	roleName                              = "extension-apiserver-authentication-reader"
+	kubeSchedulerClusterRoleName          = "system:kube-scheduler"
+	volumeSchedulerClusterRoleName        = "system:volume-scheduler"
+	webhookName                           = "kube-scheduler.scheduling.gardener.cloud"
+	volumeMountPathConfig                 = "/var/lib/kube-scheduler-config"
+)
+
+// KubeScheduler contains functions for a kube-scheduler deployer.
+type KubeScheduler interface {
+	component.DeployWaiter
+}
+
+// New creates a new instance of DeployWaiter for the kube-scheduler.
+// It requires Seed cluster with version 1.18 or 1.19.
```

Comment:
Why? What about older versions? See https://github.com/gardener/gardener/blob/master/docs/usage/supported_k8s_versions.md#seed-cluster-versions

Reply:
Some feature gates - e.g. Istio require that you have `1.16` or newer. The `kube-scheduler` prior to 1.18 had lots of the old legacy APIs in.

Reply:
OK, agreed as it's an alpha feature gate anyways. Thanks.